lib/Client.php

```diff
@@ -113,12 +113,14 @@ public function fetchFileOrFolder($path, array $properties = null) {
 			try {
 				$instance = call_user_func_array($fetchFunction, [$path, $properties]);
 				return $instance;
-			} catch (Exception $e) {
-				if (
-					strpos($e->getMessage(), $path) === false
-					&& !$this->isErrorDoesNotExist($e)
-				) {
-					# Unexpected Exception, pass it on
+			} catch (RequestException $e) {
+				if ($e->getCode() === 404) {
+					continue;
+				}
+				$payload = json_decode($e->getMessage(), true);
+				$responseCodeJson = $payload['error'];
+				$spErrorCode = (int)explode(',', $responseCodeJson['code'])[0];
+				if (!$this->isErrorDoesNotExist($spErrorCode)) {
 					throw $e;
 				}
```

Comment:
Just so it is easier to read.

```suggestion
				if ($this->isErrorDoesNotExist($spErrorCode) {
					continue;
				}
				throw $e;
```

Reply:
agreed, thanks